### PR TITLE
[Compute] Add patch spec for SharedImageGallery

### PR DESCRIPTION
**Pull request by jerryc3**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by jerryc3** [comment]
The potential new ARM errors are all expected given that
1) patch resource-> id and type are not required
2) examples would be redundant since PUT already contains examples. It would be harder to maintain if we added examples for PATCH in addition to PUT.

**Comment by lirenhe** [comment]
@jerryc3, could you take a look at the errors in LintDiff check? Let me know if you have any questions.

**Comment by jerryc3** [comment]
> @jerryc3, could you take a look at the errors in LintDiff check? Let me know if you have any questions.

HI Renhe (@lirenhe), I looked at the errors in LintDiff and I believe they are expected.

1) patch resource-> id and type should not be required for a Patch. A user should be able to pass just tags in and update the resource.
2) examples would be redundant since PUT already contains examples. It would be harder to maintain if we added examples for PATCH in addition to PUT since those examples would be duplicated.

(I followed how compute.json is defining Patch)

**Comment by lirenhe** [comment]
> @jerryc3, could you take a look at the errors in LintDiff check? Let me know if you have any questions.
> 
> HI Renhe (@lirenhe), I looked at the errors in LintDiff and I believe they are expected.
> 
> patch resource-> id and type should not be required for a Patch. A user should be able to pass just tags in and update the resource.
> examples would be redundant since PUT already contains examples. It would be harder to maintain if we added examples for PATCH in addition to PUT since those examples would be duplicated.
> 
> (I followed how compute.json is defining Patch)

Do check: https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#r2020
Please note, id & type are readonly properties, user does not need to provide those values.

The example should be a good start point for user to understand how to use the your API and it would be part of the document. So please provide the examples.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/gallery.json [hunk 1]
```diff
@@ -101,6 +101,55 @@
...
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/GalleryUpdate"
+            },
+            "description": "Parameters supplied to the update Shared Image Gallery operation."
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (lirenhe) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/gallery.json]: For long running operation, why the response code is 200?

Comment (jerryc3) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/gallery.json]: This was a decision made by all the compute RPs. In the past 202 would have been returned but for a year now all the CompuuteRPs return 200 and the user polls via location/azyre-async polling.

**Comment by jerryc3** [comment]
@koralski added Patch for GalleryApp(Verison) as well

**Review by lirenhe — APPROVED** [review]

**Comment by ravbhatnagar** [comment]
signing off from ARM side.

**Pull request merged**